fromOkayToGreat/gt-anthro-library-next
# 5 implement site colors in tailwind configuration

## :sparkles: Enhance Routing and Styling Structure

### 📰 Description

This PR introduces a significant restructuring of the project directories and files to resolve issues with global styling not being applied correctly. By moving page components from the `pages/` directory to the `app/` directory and its subdirectories, we've taken a more manual approach to routing and styling, ensuring a consistent and maintainable setup.

### 🌍 Changes Made

- **Project Restructure**: Moved page components from `pages/` to `app/` and organized them into appropriate subdirectories.
- **Global Styles Application**: Ensured global styles are correctly applied by adjusting the placement and reference of layout and CSS files.

### 💡 Rationale

The previous structure, while conventional, wasn't applying global styles as expected due to Next.js's specific expectations around the `pages/_app.tsx` file and routing. By reorganizing the files and explicitly defining the structure, we've gained more control over the styling and routing, leading to a more consistent and predictable application behavior.

### 🔍 How to Test

1. Pull the changes from this branch.
2. Run the application locally.
3. Navigate through various pages to ensure styles are applied consistently.
4. Review the code changes to understand the new structure and setup.

### 📸 Screenshots

![testing color on home page](https://github.com/fromOkayToGreat/gt-anthro-library-next/assets/88689386/edd3cc87-a209-4360-a485-136367d5c19f)

### 📚 Additional Information

There will be a related issue that will test the hypothesis:

`Custom App Component: In Next.js, pages/_app.tsx is a special file that allows you to initialize pages. It's the place to add global CSS and wrap all your pages with layout components. If you don't have a pages/_app.tsx but have an app/layout with global CSS, the global styles might not be applied correctly because Next.js isn't aware of how you want to structure your application globally.`



---

### ✅ Definition of Done

- [x] All pages are correctly routed and render without styling issues.
- [x] Global styles are consistently applied across the application.
- [x] The project structure is logical, maintainable, and well-documented.
